### PR TITLE
Add CrossVersionTokenValidation test project

### PR DESCRIPTION
**Pull request by ChenYan98**

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs [hunk 1]
```diff
@@ -553,6 +553,8 @@ protected virtual SamlSubject CreateSubject(SecurityTokenDescriptor tokenDescrip
                 //    }
             }
 
+            samlSubject.ConfirmationMethods.Add(SamlConstants.BearerConfirmationMethod);
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs]: please remove the commented code.
thanks.

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs [hunk 2]
```diff
@@ -1006,9 +1008,9 @@ protected virtual void ValidateConditions(SamlSecurityToken securityToken, Token
                 {
                     SamlAudienceRestrictionCondition audienceRestriction = condition as SamlAudienceRestrictionCondition;
                     if (validationParameters.AudienceValidator != null)
-                        validationParameters.AudienceValidator(audienceRestriction.Audiences, securityToken, validationParameters);
+                        validationParameters.AudienceValidator(audienceRestriction.Audiences.ToDictionary(x => x.ToString()).Keys, securityToken, validationParameters);
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs]: Best to use x.OriginalString() as that leaves the value unchanged.

File: test/CrossVersionTokenValidation.Tests/SamlCrossVersionTokenValidationTests.cs [hunk 3]
```diff
@@ -0,0 +1,789 @@
...
+            {
+                context.Diffs.Add(Environment.NewLine + "obj1 is null");
+                return false;
+            }
+
+            if (objs2 == null)
+            {
+                context.Diffs.Add(Environment.NewLine + "obj2 is null");
+                return false;
+            }
```
Comment (brentschmaltz) [inline test/CrossVersionTokenValidation.Tests/SamlCrossVersionTokenValidationTests.cs]: You could break out these three checks into a separate function. then each method could use them.

File: buildConfiguration.xml [hunk 4]
```diff
@@ -14,6 +14,7 @@
       <project name="Microsoft.IdentityModel.Protocols.OpenIdConnect" />
     </src>
     <test>
+	  <project name="CrossVersionTokenValidation.Tests" test="yes"/>
```
Comment (ChenYan98) [inline buildConfiguration.xml]: fixed

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs (hunk 1 above)
Comment (ChenYan98) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs]: fixed

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs (hunk 2 above)
Comment (ChenYan98) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs]: Fixed

File: test/CrossVersionTokenValidation.Tests/SamlCrossVersionTokenValidationTests.cs (hunk 3 above)
Comment (ChenYan98) [inline test/CrossVersionTokenValidation.Tests/SamlCrossVersionTokenValidationTests.cs]: If I understand right, then it is the same thing like add ContinueCheckingEquality. If it is, then fixed.

**Comment by brentschmaltz** [comment]
:shipit:

**Pull request merged**